geoh5py/ui_json/validation.py

```diff
@@ -51,17 +46,54 @@ class InputValidation:
 
     def __init__(
         self,
-        validations: dict[str, Any] | None,
+        validators: dict[str, BaseValidator] = None,
+        validations: dict[str, Any] | None = None,
         ignore_list: tuple = (),
         ignore_requirements: bool = False,
         workspace: Workspace = None,
+        ui_json: dict[str, Any] = None,
     ):
-        self._validators: dict[str, BaseValidator] = {}
+        self.validations: dict[str, Any] | None = validations
+        self.ui_json: dict[str, Any] = ui_json
+        self.validators: dict[str, BaseValidator] = validators
         self.workspace: Workspace | None = workspace
-        self.validations = validations
         self.ignore_list: tuple = ignore_list
         self.ignore_requirements: bool = ignore_requirements
 
+    @property
+    def ui_json(self):
+        return self._ui_json
```

Comment:
Propose to NOT store ui_json, but just call the `infer_validations` method directly in the init. Just scared of having the ui_json lingering in many classes...

Reply:
ok, I've eliminated the setting and added a raise attribute error on the getter.